### PR TITLE
Flip swap angle sign in iswap

### LINKED ISSUE #1882: Handedness between ISwapPowGate and FSim is different.
The ISwapPowGate and FSim gate rotate in different directions on the |01>, |10> subspace.  This feels a bit odd to me.

### PR DESCRIPTION
**Pull request by viathor**
Fixes #1882.

### CODE REVIEW

File: cirq/ops/common_gates.py [hunk 1]
```diff
@@ -1081,14 +1081,12 @@ class ISwapPowGate(eigen_gate.EigenGate,
...
             (0, np.diag([1, 0, 0, 1])),
-            (+0.5, np.array([[0, 0, 0, 0],
-                             [0, 0.5, 0.5, 0],
-                             [0, 0.5, 0.5, 0],
-                             [0, 0, 0, 0]])),
-            (-0.5, np.array([[0, 0, 0, 0],
-                             [0, 0.5, -0.5, 0],
-                             [0, -0.5, 0.5, 0],
-                             [0, 0, 0, 0]])),
+            (-0.5,
```
Comment (vtomole) [inline cirq/ops/common_gates.py]: Make the arrays more readable by calling `#yapf disable` around these parts.

Comment (viathor) [reply cirq/ops/common_gates.py]: Done.

File: cirq/ops/common_gates_test.py [hunk 2]
```diff
@@ -620,13 +620,12 @@ def test_iswap_str():
 
 
 def test_iswap_unitary():
```
Comment (vtomole) [inline cirq/ops/common_gates_test.py]: I'm seeing different answers on https://quantumcomputing.stackexchange.com/questions/2594/what-is-the-matrix-of-the-iswap-gate. Some people say it should be `+i` while some people argue that it doesn't matter if it's `+i` or `-i`. Could someone please clear this up for me?

Comment (viathor) [reply cirq/ops/common_gates_test.py]: There are two conflicting conventions. I think we have decided to use `-i` and already switched fSim in #1726. However, ISwap was missed in that PR and so @dabacon filed #1882. This PR fixes the inconsistency and completes our switch to the `-i` convention for both gates.

File: cirq/ops/common_gates.py [hunk 3]
```diff
@@ -1060,46 +1060,48 @@ class ISwapPowGate(eigen_gate.EigenGate,
...
          [0, 0, 0, 1]]
 
     where:
 
         c = cos(π·t/2)
-        s = sin(π·t/2)
+        s = -i·sin(π·t/2)
 
     `cirq.ISWAP`, the swap gate that applies i to the |01> and |10> states,
     is an instance of this gate at exponent=1.
```
Comment (cduck) [inline cirq/ops/common_gates.py]: Is the sign of i wrong here now?  This was just changed from -1 a few days ago in #1848.

File: cirq/ops/common_gates_test.py [hunk 4]
```diff
@@ -620,13 +620,12 @@ def test_iswap_str():
...
-    cirq.testing.assert_allclose_up_to_global_phase(
-        cirq.unitary(cirq.ISWAP),
-        np.array([[1, 0, 0, 0],
-                  [0, 0, 1j, 0],
-                  [0, 1j, 0, 0],
-                  [0, 0, 0, 1]]),
-        atol=1e-8)
+    cirq.testing.assert_allclose_up_to_global_phase(cirq.unitary(cirq.ISWAP),
+                                                    np.array([[1, 0, 0, 0],
+                                                              [0, 0, -1j, 0],
```
Comment (Strilanc) [inline cirq/ops/common_gates_test.py]: I think this is wrong. The definition of an iswap is to put i's along the anti-diagonal. I carefully looked into this and asked several people about it before putting this matrix into the test.

See this quantum stackexchange question: https://quantumcomputing.stackexchange.com/a/2596/119

File: cirq/ops/common_gates.py (hunk 1 above)
Comment (dabacon) [reply cirq/ops/common_gates.py]: I think also you can add a trailing ","?  
`[[0, 0, 0, 0], 
 [0, 0.5, 0.5, 0],
 [0, 0.5, 0.5, 0],
 [0, 0.5, 0.5, 0],
 [0, 0, 0, 0],]`

File: cirq/ops/common_gates.py (hunk 3 above)
Comment (dabacon) [reply cirq/ops/common_gates.py]: I think we want this to be consistent with FSim.  #1882

**Comment by mpharrigan** [comment]
The only consistent thing is inconsistency

**Comment by viathor** [comment]
We've decided to keep opposite sign conventions in iSWAP and fSim.

**Pull request closed without merging**